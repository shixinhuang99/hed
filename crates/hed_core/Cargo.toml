[package]
name = "hed_core"
version = "0.1.0"
edition = "2021"

[lib]
doctest = false

[dependencies]
thiserror = "1.0.63"
