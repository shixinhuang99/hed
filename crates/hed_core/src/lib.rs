pub mod channel;
pub mod error;
mod hed;
mod hosts_info;

pub use hed::Hed;
pub use hosts_info::HostsInfo;
