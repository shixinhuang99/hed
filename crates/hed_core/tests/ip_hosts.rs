use hed_core::IpHosts;
use insta::assert_debug_snapshot;

#[test]
fn test_ip_hosts_parse() {
	let s = "172.16.254.1 a.com b.com b.com # comment";
	assert_debug_snapshot!("ipv4", s.parse::<IpHosts>().unwrap());
}

#[test]
fn test_ip_hosts_parse_ipv6() {
	let s =
		"2402:1200:4f00:1234:0000:5678:9abc:def0 a.com b.com b.com # comment";
	assert_debug_snapshot!("ipv6", s.parse::<IpHosts>().unwrap());
}
