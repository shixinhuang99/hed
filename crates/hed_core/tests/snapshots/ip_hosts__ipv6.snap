---
source: crates/hed_core/tests/ip_hosts.rs
expression: "s.parse::<IpHosts>().unwrap()"
---
IpHosts {
    ip: "2402:1200:4f00:1234:0000:5678:9abc:def0",
    hosts: {
        "a.com",
        "b.com",
    },
}
